src/vuln_analysis/tools/osv_tools.py

```
@@ -0,0 +1,202 @@
+@lru_cache(maxsize=256)
+def _fetch_osv(cve_id: str) -> dict | None:
+    url = f"{_OSV_API_BASE}/{cve_id}"
+    logger.info("osv_lookup: fetching %s", url)
+    with httpx.Client(timeout=30, follow_redirects=True) as client:
+        resp = client.get(url)
+        if resp.status_code == 404:
+            return None
+        resp.raise_for_status()
+        data = resp.json()
+        if not isinstance(data, dict):
+            raise ValueError("OSV response is not a JSON object")
```

Ruff (0.16.1)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)